### ISSUE TITLE
Add pagination to list notes, accounts, and nullifiers

### ISSUE THREAD

**Issue by hackaugusto** [0xPolygonMiden/miden-node#163]
The current version of the endpoints to list notes, accounts, and nullifiers does not have pagination. The [best practice](https://protobuf.dev/programming-guides/api/#define-pagination-api) is to return a continuation token.

**Comment by bobbinth** [comment]
An alternative is to remove these endpoints altogether. They were meant solely for testing, but I'm not sure we are using them.

cc @Mirko-von-Leipzig.

**Comment by Mirko-von-Leipzig** [comment]
If they're not in use then I'm inclined to delete them :). Adding pagination via a continuation token would also be rather simple to implement though.

I assume the wallet/client use the `rpc` server list, and don't directly access the "internal" store api @igamigo?

**Comment by igamigo** [comment]
> I assume the wallet/client use the `rpc` server list, and don't directly access the "internal" store api @igamigo?

Correct. These endpoints have been useful in the past for debugging but I don't think they've been used as of recently so I agree it would be good to delete them.

**Comment by Mirko-von-Leipzig** [comment]
Closed by #608
